src/main/java/com/github/cameltooling/lsp/internal/completion/CamelKModelineOptionNames.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cameltooling.lsp.internal.completion;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.lsp4j.CompletionItem;
+
+public class CamelKModelineOptionNames {
+	
+	private CamelKModelineOptionNames() {}
+
+	private static final List<CompletionItem> COMPLETION_ITEMS;
+	
+	static {
```

Comment:
I am wondering if we can grab possible keys in a more dynamic way.

Reply:
hardcoded list for now. The "useful" ones, as mentioned in Camel K
documentation, are not available dynamically for now anyway.

see https://issues.redhat.com/browse/FUSETOOLS2-516 that would allow a dynamic retrieval of catalog.